src/peer_id.rs

```diff
@@ -16,32 +17,32 @@ pub struct PeerId {
     public_key: PublicKey,
 }
 
-impl PeerId {
-    /// Create a new PeerId from a public key
-    pub fn from_public_key(public_key: PublicKey) -> PeerId {
-        PeerId { public_key }
-    }
+#[derive(Clone, Debug, Eq, PartialEq, Hash, PartialOrd, Ord)]
+pub struct PeerNetID<T>(pub T)
```

Comment:
Why do you need this wrapper structure ? The function can't just take the type T ?